lib/AdobeMarketingCloud/Api/Report.php

```diff
@@ -0,0 +1,183 @@
+     *
+     * @param string $reportSuiteID
+     * @param bool $returnAsIdArray
+     *
+     * @return array
+     */
+    public function getElements($reportSuiteID, $returnAsIdArray = false)
+    {
+        $response = $this->post('Report.GetElements', array('reportSuiteID' => $reportSuiteID));
+        if ($returnAsIdArray) {
+            $filtered = array();
+            foreach ($response as $element) {
+                $filtered[$element['id']] = $element['name'];
+            }
+            return array_unique($filtered);
+        }
+        return $this->returnResponse($response);
+    }
+
+    /**
+     * Retrieve events for your report suites
+     *
+     * @param string $reportSuiteID
+     * @param bool $returnAsIdArray
+     *
+     * @return array
+     */
+    public function getMetrics($reportSuiteID, $returnAsIdArray = false)
+    {
+        $response = $this->post('Report.GetMetrics', array('reportSuiteID' => $reportSuiteID));
+        if ($returnAsIdArray) {
+            $filtered = array();
+            foreach ($response as $metric) {
+                $filtered[$metric['id']] = $metric['name'];
+            }
+            return array_unique($filtered);
+        }
+        return $this->returnResponse($response);
+    }
+
+    /**
+     * Submits a report request.
+     *
+     * https://marketing.adobe.com/developer/documentation/analytics-reporting-1-4/r-queue
+     *
+     * @param   array $reportDescription  the report description array
+     * @return  array
+     *   - reportID: The report ID of the queued report
+     */
+    public function queue($reportDescription)
+    {
+        $response = $this->post('Report.Queue', array(
+            'reportDescription' => $reportDescription,
+        ));
+
+        return $this->returnResponse($response);
+    }
+
+    /**
+     * Gets a report by report ID
+     * https://marketing.adobe.com/developer/documentation/analytics-reporting-1-4/r-get
+     *
+     * @param   mixed     The report ID or report description
+     * @return  tns:report      A structure containing the report data for the specified reportID
+     */
+    public function getReport($report)
+    {
+        if (is_numeric($report)) {
+            $response = $this->post('Report.Get', array(
+                'reportID' => $report,
+            ));
+            $report = $this->returnResponse($response);
+        } else {
+            $report = $this->queueAndGetReport($report);
+        }
+
+        return $report;
+    }
+
+    /**
+     * Cancels a previously submitted report request, and removes it from the processing queue.
+     * https://marketing.adobe.com/developer/documentation/analytics-reporting-1-4/r-cancel
+     *
+     * @param   int     The report ID returned as part of the "queue report" request.
+     * @return  int     Returns 1 if the operation is successful, or 0 if the operation failed.
+     */
+    public function cancelReport($reportId)
```

Comment:
might just wanna call this method `cancel()` to make it consistent with `Report.Queue` => `queue()`